portfolio/src/main/webapp/script.js

```diff
@@ -65,3 +65,186 @@ function addRandomFunFact() {
+ */
+function createPageElement(text, active) {
+  const pageElement = document.createElement('li');
+  pageElement.className = 'page-item';
+  if (active) { 
+    pageElement.classList.add('active'); 
+  }
+  
+  const linkElement = document.createElement('a');
+  linkElement.className = 'page-link';
+  linkElement.innerText = text;
+  pageElement.appendChild(linkElement);
+
+  pageElement.addEventListener('click', event => changePage(event));
+  return pageElement;
+}
+
+/**
+ * Changes the active page element and displays the comments on that page when a page element is clicked.
+ * @param {!Event} event The click event that triggers the change of page.
+ */
+function changePage(event) {
+  const currentPage = document.querySelector('.page-item.active');
+  const pageText = event.currentTarget.firstElementChild.innerText;
+  
+  if (pageText == 'Previous') {
+    // If the currentPage is the first page, then return without change.
+    if (currentPage.firstElementChild.innerText == 1) { 
+      return; 
+    }
+
+    currentPage.previousSibling.classList.add('active');
+  } else if (pageText == 'Next') {
+    // If the currentPage is the last page, then return without change.
+    const numPages = document.querySelectorAll('#page-list li').length;
+    if (currentPage.firstElementChild.innerText == numPages - 2) { 
+      return; 
+    }
+
+    currentPage.nextSibling.classList.add('active');
+  } else {
+    event.currentTarget.classList.add('active');
+  }
+
+  currentPage.classList.remove('active');
+
+  // Fetch comments on that page without recreating pagination.
+  fetchComments(false);
+}
+
+/**
+ * Fetches the comments from the data server and displays the comments on the currently active page.
+ * @param {boolean=} createNewPagination Whether a new pagination bar should be created.
+ */
+function fetchComments(createNewPagination = false) {
+  fetch('/data').then(response => response.json()).then(comments => {
+    const commentsListElement = document.getElementById('comments-list');
+    commentsListElement.innerHTML = '';
+
+    const commentsPerPageElement = document.getElementById('num-comments-per-page');
+    const commentsPerPage = commentsPerPageElement.options[commentsPerPageElement.selectedIndex].value;
+
+    const currentPage = document.querySelector('.page-item.active');
+    var currentPageNumber = currentPage.firstElementChild.innerText;
+    if (createNewPagination) {
+      currentPageNumber = createPaginationBar(Math.ceil(comments.length / commentsPerPage), currentPageNumber);
+    }
+
+    const startIndex = (currentPageNumber - 1) * commentsPerPage;
+    const endIndex = Math.min(Number(commentsPerPage) + Number(startIndex), comments.length);
+    for (var i = startIndex; i < endIndex; i++) {
+      commentsListElement.appendChild(createCommentElement(comments[i]));
+    }
+  });
+}
+
+/**
+ * Creates an <li> element containing a comment, including the username, content, time, and delete button.
```

Comment:
I would split this to make it more clear.

```suggestion
 * Creates an <li> element containing an individual comment. The element includes the following attributes associated with the comment: username, content, time, and delete button.
```

Reply:
Got it